Sources/Fullscreen/EasterEggs/DrumMachine/DrumMachineView.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  Copyright © 2018 FINN.no. All rights reserved.
+//
+
+import UIKit
+
+public final class DrumMachineView: UIView {
+    private let padSpacing: CGFloat = .largeSpacing
+    private let cellSpacing: CGFloat = .mediumSpacing
+    private let numberOfPads = 16
+    private var currentPad = 0
+    private var timer: Timer?
+    private var beatsPerMinute: Float = 120
+
+    private lazy var compositions: [Instrument: [Bool]] = self.makeEmptyCompositions()
+
+    private var instrument: Instrument = .kick {
+        didSet {
+            collectionView.reloadData()
+        }
+    }
+
+    private var timeInterval: Double {
+        return (60.0 / Double(beatsPerMinute)) / 4
+    }
+
+    private lazy var selectorView: InstrumentSelectorView = {
+        let selectorView = InstrumentSelectorView(instrument: self.instrument)
+        selectorView.translatesAutoresizingMaskIntoConstraints = false
+        return selectorView
+    }()
+
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: self.collectionViewLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.backgroundColor = .black
+        collectionView.register(DrumPadCollectionViewCell.self)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        return collectionView
+    }()
+
+    private lazy var collectionViewLayout: UICollectionViewLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        layout.minimumInteritemSpacing = cellSpacing
+        layout.minimumLineSpacing = cellSpacing
+        layout.sectionInset = UIEdgeInsets(top: padSpacing, left: padSpacing, bottom: padSpacing, right: padSpacing)
+        return layout
+    }()
+
+    private lazy var bpmSlider: UISlider = {
```

Comment:
Could we rename `bpmSlider` to `beatsPerMinuteSlider`? Is that what stands for? 😁